docs/website/contents/for-developers/utxo-hd/utxo-hd.md

```diff
@@ -0,0 +1,434 @@
+  prohibitely costly.
+
+- Make `Value (LedgerState (HardForkBlock xs))` a sum type that can hold values of
+  any eras. This solution makes it very easy to carry `LedgerTables` in the
+  Consensus layer as values do not need to be translated, in fact
+  values from older eras might co-exist with those of the current one. The
+  disadvantage of this solution is that injecting the ledger tables in the
+  ledger state (so `withLedgerTables :: LedgerTables ... mk -> LedgerState ... anymk -> LedgerState ... mk`)
+  implies that we are going from hard-fork keys and values to keys and values of
+  the particular era, making the necessary era translations on-the-fly.
+
+  This tradeoff was considered acceptable and because of it we put much care
+  in only injecting small tables, such as the set of values needed to apply a
+  block (which is bound by the maximum size of the block). Developers integrating
+  the UTXO solution in other tools should understand this limitation and put
+  great care in not violating it for example by injecting and projecting the whole
+  UTXO set on every block which would simply blow up the memory consumption.
+
+It is important to note that for any era in the Cardano blockchain, the `EraCrypto`
+type family instance is the same (`StandardCrypto`), which makes all `TxIn (EraCrypto era)` keys equal. Thanks
+to this, we can define the `Key` for `HardForkBlocks` equal to this same type,
+what we call a `CanonicalTxIn`.
+
+### Storing snapshots
+
+Before UTXO-HD, ledger state snapshots were CBOR-serialized files containing a full
+`ExtLedgerState blk` value. Now there is a separation between the `ExtLedgerState blk EmptyMK` file and the `LedgerTables (ExtLedgerState blk) ValuesMK`. This means that snapshots from before UTXO-HD are
+incompatible with the UTXO-HD design and replaying the chain will be needed when
+enabling UTXO-HD. Moreover, snapshots created when using one of the UTXO-HD backends
+cannot be used with the other backend, and will require a replay.
+
+| | `ExtLedgerState blk EmptyMK` | `LedgerTables (ExtLedgerState blk) ValuesMK` | Live tables |
+|--|--|--|--|
+| In-memory | `<db-root>/ledger/<slotno>/state` | `<db-root>/ledger/<slotno>/state/tables/tvar` | N/A |
+| On-disk   | `<db-root>/ledger/<slotno>/state` | `<db-root>/ledger/<slotno>/state/tables/data.mdb` | `<db-root>/ledgerdb/data.mdb` |
+
+In the tables part of the snapshot, the in-memory backend will store a serialization of the `Map (Key (CardanoBlock c)) (Value (CardanoBlock c))`, whereas the on-disk backend will store a copy of the LMDB database.
+
+## Impact on the node
+
+The in-memory backend should have very little impact in the node.
+
+The cardano-node will perform two operations on startup, and each of them suffer a varying impact for the on-disk backend:
+
+| | Impact | Approximated time difference |
+|--|--|--|
+| Syncing | Low, cryptographic operations dominate the performance | 16h vs 17h |
```

Comment:
can this last column be split into two, one for each memory backend?

Reply:
This table is only for the on-disk backend. I made the `on-disk` part bold to make it clearer.